mandos-go/expression/reconstructor/reconstructor.go

```diff
@@ -55,12 +58,19 @@ func unknownByteArrayPretty(bytes []byte) string {
 		return "[]"
 	}
 
+	// fully interpret as string
 	if canInterpretAsString(bytes) {
-		return fmt.Sprintf("0x%s (``%s)", hex.EncodeToString(bytes), string(bytes))
+		return fmt.Sprintf("0x%s (str:%s)", hex.EncodeToString(bytes), string(bytes))
+	}
+
+	// interpret as number
+	if len(bytes) < maxBytesInterpretedAsNumber {
```

Comment:
Shouldn't the interpretation as number have priority over the interpretation as string?